pkg/backend/runtime.go

```diff
@@ -43,3 +44,16 @@ type ContainerRuntime interface {
 	// The snapshot key must also be saved in the returned map with the key "FakeMetaDataSnapshotKey".
 	ListSnapshots(ctx context.Context) ([]SnapshotMetadata, error)
 }
+
+// Mounter is a generic interface used for mounting images
+type Mounter interface {
+	// Mount mounts a specific image
+	Mount(
+		ctx context.Context, volumeId string, target MountTarget, image docker.Named, ro bool) (err error)
+
+	// Unmount unmounts a specific image
+	Unmount(ctx context.Context, volumeId string, target MountTarget) error
+
+	// ImageExists checks if the image already exists on the local machine
+	ImageExists(ctx context.Context, image docker.Named) bool
```

Comment:
I would like to mentioned that I'm not too well versed with the codebase, so this question might not have its place in this particular PR's review.

I can ask this question separately as a discussion, if it doesn't make sense here.

Reply:
@vadasambar

Reply:
> Is there a reason to use the interface `docker.Named` instead of `containerd#Image`?

Might be historical reason.

Reply:
> I can ask this question separately as a discussion, if it doesn't make sense here.

We use `docker` package at multiple places. It would be good to evaluate if we can move to something more generic i.e., something not specific to a CRI implementation.